server/lobby.js

```diff
@@ -227,7 +228,7 @@ class Lobby {
         this.sockets[ioSocket.id] = socket;
 
         if(socket.user) {
-            this.users[socket.user.username] = socket.user;
+            this.users[socket.user.username] = Util.getUserWithDefaultsSet(socket.user);
```

Comment:
Is there a more central place that this can be done, rather than sprinkle it here and in `Server`? Seems like having it defaulted properly once already on the socket object would be less error prone.